Remove grunt

simison opened the pull request:
1. Uses wp-i18n directly via [`node-wp-i18n`](https://github.com/cedaro/node-wp-i18n) module instead of [Grunt wrapper](https://github.com/cedaro/grunt-wp-i18n). See https://github.com/cedaro/grunt-wp-i18n/issues/61#issuecomment-194076693

    Allows removing grunt task runner. :tada: 🎈 


2. Updates `wp-i18n` from quite old version to the latest. (`grunt-wp-i18n@0.4.6` [is from July 2014](https://github.com/cedaro/grunt-wp-i18n/releases?after=v0.4.7))

3. Removes POT generation, we're not using it anymore - GlotPress at WordPress.org generates a POT file based on existing strings that we ship with the plugin.


## Test
1. Grab this branch and `yarn install`
1. Test if this still works as expected:

    ```bash
    yarn add-textdomain
    ```

Closes https://github.com/Automattic/jetpack/issues/8882

oskosk commented:
I tested building the pot file with master and this branch and the output is different. Checking now if there's anything significant. So far things like this are differing:

```
 #: 3rd-party/debug-bar/class.jetpack-search-debug-bar.php:26
 #: docker/wordpress/wp-content/plugins/jetpack-dev/3rd-party/debug-bar/class.jetpack-search-debug-bar.php:26
@@ -6931,8 +6931,8 @@ msgstr ""
 #: docker/wordpress/wp-admin/edit-comments.php:185
 msgid ""
 "<a "
-"href=\"https://codex.wordpress.org/Administration_Screens#Comments\">"
-"Documentation on Comments</a>"
+"href=\"https://codex.wordpress.org/Administration_Screens#Comments\""
+">Documentation on Comments</a>"
 msgstr ""
```

simison commented:
@oskosk this updates `wp-i18n` from quite an old version to the latest (`grunt-wp-i18n@0.4.6` [was from July 2014](https://github.com/cedaro/grunt-wp-i18n/releases?after=v0.4.7)) so that might be causing some differences, too. I updated PR description.

You could confirm by upgrading it in master and comparing results: `yarn add grunt-wp-i18n@~1.0.2 --save-dev`

simison commented:
I think the most notable diff is this:
```diff
-"Project-Id-Version: Jetpack by WordPress.com 3.7.0-beta4\n"
-"Report-Msgid-Bugs-To: http://wordpress.org/support/plugin/jetpack\n"
-"POT-Creation-Date: 2015-09-03 08:10:56+00:00\n"
+"Project-Id-Version: Jetpack by WordPress.com 6.1-alpha\n"
+"Report-Msgid-Bugs-To: https://wordpress.org/support/plugin/jetpack\n"
+"POT-Creation-Date: 2018-04-03 21:14:37+00:00\n"
```

...and that's why rest of the pot file differs so much.

oskosk commented:
This one needs a rebase!

simison commented:
Let's first figure out if these Grunt tasks are needed at all and if so, does the latest version of `wp-i18n` work for us.

simison commented on package.json:
```diff
@@ -37,8 +37,8 @@
     "test-client": "NODE_ENV=test NODE_PATH=tests:_inc/client:node_modules/@automattic/dops-components/client tests/runner.js",
-    "add-textdomain": "grunt addtextdomain",
-    "build-pot": "grunt makepot",
+    "add-textdomain": "wpi18n addtextdomain --textdomain=jetpack --glob-pattern='!(docker|node_modules|tests|tools|vendor){*.php,**/*.php}'",
```
`wpi18n@1.2.0` now supports `--exclude` for `addtextdomain` command so those ignored folders from glob pattern could be moved out.

That would be just for readability though, `glob-pattern` already works as expected.

zinigor commented:
Thanks for doing this, let's get rid of Grunt! As for the actual tasks - we can get rid of POT generation, we're not using it anymore - GlotPress at WordPress.org generates a POT file based on existing strings that we ship with the plugin. That's why we haven't updated the included file in a while - since plugin language packs became available in WordPress.

As for textdomain fixes - we can probably run it as a precommit check, or as part of the build process.

simison commented:
@zinigor @oskosk I rebased this and removed the`build-pot` script.

Here it is just for the record:
```
"build-pot": "wpi18n makepot --domain-path=/languages --main-file=jetpack.php --pot-file=jetpack.pot --exclude=docker,node_modules,tests,tools,vendor && wpi18n info",
```

simison commented:
>we can get rid of POT generation, we're not using it anymore - GlotPress at WordPress.org generates a POT file based on existing strings that we ship with the plugin. That's why we haven't updated the included file in a while - since plugin language packs became available in WordPress.

Could we remove something from the `/languages` folder?

brbrr commented:
Hey @zinigor 
I think you might have missed the comment above.

zinigor commented:
Whoops! Sorry!

The only thing that ***I think*** is not used in the languages folder right now is the `jetpack.pot` file. We either need to update it or stop shipping it. All other language files are needed - we update existing translations for every release, and remove `po` and `mo` files for languages that have language packs available from WordPress community.

simison commented:
⚡️⚡️⚡️🚀 🤘😍👍🚀⚡️⚡️⚡️
